### ISSUE TITLE
Unable to perform more than two statuses.filter on a Streaming API

### ISSUE THREAD

**Issue by godlike64** [sixohsix/twitter#318]
First of all, I've guided myself via the examples and mainly through trial and error, so feel free to categorize this as properly (or close if it's not a bug).

What I want to do:
- I'm currently tinkering with websockets, using the twitter API to parse streams and show them in real time (this is for a bigger project). The websockets part is done for.
- I'm using a personal OAuth token for testing purposes.
- The idea is to be able to parse multiple filters from the same public stream

The code (currently running on Python 3.5 - I need this for async reasons but I can try this out on earlier Python versions if necessary):

~~~
auth = OAuth([...])
stream = TwitterStream(auth=auth)
query_args = dict()
query_args['track'] = 'python'
query_args2 = dict()
query_args2['track'] = 'devops'
query_args3 = dict()
query_args3['track'] = 'linux'
filter1 = stream.statuses.filter(**query_args)
filter2 = stream.statuses.filter(**query_args2)
filter3 = stream.statuses.filter(**query_args3)
~~~

The third filter() call throws the following:

~~~
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.5/site-packages/twitter/api.py", line 308, in __call__
    return self._handle_response(req, uri, arg_data, _timeout)
  File "/usr/lib64/python3.5/site-packages/twitter/stream.py", line 285, in _handle_response
    _timeout or timeout, heartbeat_timeout)
  File "/usr/lib64/python3.5/site-packages/twitter/stream.py", line 209, in handle_stream_response
    raise TwitterHTTPError(e, uri, 'json', arg_data)
  File "/usr/lib64/python3.5/site-packages/twitter/api.py", line 72, in __init__
    data = json.loads(data.decode('utf8'))
  File "/usr/lib64/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
~~~

Cryptic error, so I went to /usr/lib64/python3.5/site-packages/twitter/stream.py:209 and added a print statement for that exception (huh, the exception seems not to be caugh properly, but nevermind, a separate issue I guess):

~~~
HTTP Error 420: Enhance Your Calm
~~~

Am I doing something wrong, or is it actually not possible to perform multiple filters from the same public stream? Twitter's 420 limit for the streaming API seems to be fixed at two connections per OAuth (and maybe IP address?), which is quite low if I want to keep track of, say, 3 filtered streams :).

**Comment by RouxRC** [comment]
Hi, twitter only allows to use one stream at a time, so to do what you're trying to do, you have to call the stream only once with all three keywords in the track argument:
```
filter1 = stream.statuses.filter(track="python,devops,linux")
```

**Comment by RouxRC** [comment]
if you want to store the data separately locally, you'll have to redispatch by filtering the "text" field of the tweets yourself, including looking into the unshortened urls and the original tweet for RTs

**Comment by RouxRC** [comment]
or use three different api keysets

**Comment by godlike64** [comment]
Thanks for the clarification! So, if I want to track different keywords, I have to use different OAuth tokens. Although, it does seem that twitter allows me two connections per token, but nevermind.

Now, suppose I am already tracking some keywords, and wish to track some additional ones. How do I go about disconnecting the current stream and reconnecting to use the new filter? Or, if at all possible, how do I change the current filter?

Thanks again!

**Comment by RouxRC** [comment]
you cannot change the argument while connected, you need to disconnect and reconnect, so for instance something roughly like this:
```python
def run_stream(keywords):
  for result in stream.statuses.filter(track=keywords):
    do_something(result)
    if keywords_changed():
      break
  return run_stream(read_new_keywords())
```

**Comment by RouxRC** [comment]
closable?
